### PR TITLE
Refactor dev-env variable names

### PR DESCRIPTION
**Pull request by Rozzii**
The goal of this commit is to make the variable names involved in
network configuration more explicit. The main goal of this commit is to
make the difference between the "provisioner" network and the "provisioning"
network more prononced.

In the future the dev-env will support at least 2 different
network topologies thus the provisioner (e.g. ironic) won't be necessarily available
in the provisioning network the same way as it is available now.

This commit is a forruner of future changes that will introduce additional network
organization options.

List of name changes:

 - CLUSTER_PROVISIONING_INTERFACE -> BARE_METAL_PROVISIONER_INTERFACE
 - PROVISIONING_IP -> BARE_METAL_PROVISIONER_IP
 - PROVISIONING_IPV6 -> BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY
 - PROVISIONING_NETWORK -> BARE_METAL_PROVISIONER_NETWORK
 - PROVISIONING_CIDR -> BARE_METAL_PROVISIONER_CIDR
 - PROVISIONING_NETMASK -> BARE_METAL_PROVISIONER_NETMASK
 - CLUSTER_PROVISIONING_IP -> CLUSTER_BM_PROVISIONER_IP
 - PROVISIONING_URL_HOST -> BARE_METAL_RPOVISIONER_URL_HOST
 - CLUSTER_URL_HOST -> CLUSTER_BARE_METAL_PROVISIONER_HOST
 - VIRSH_DHCP_V4_START -> EXTERNAL_DHCP_V4_START
 - VIRSH_DHCP_V4_END -> EXTERNAL_DHCP_V4_START
 - BAREMETALV4_POOL_RANGE_START -> IPAM_EXTERNALV4_POOL_RANGE_START
 - BAREMETALV4_POOL_RANGE_END -> IPAM_EXTERNALV4_POOL_RANGE_END
 - VIRSH_DHCP_V6_START ->  EXTERNAL_DHCP_V6_START
 - VIRSH_DHCP_V6_END -> EXTERNAL_DHCP_V6_END
 - BAREMETALV6_POOL_RANGE_START ->  IPAM_EXTERNALV6_POOL_RANGE_START
 - BAREMETALV6_POOL_RANGE_END -> IPAM_EXTERNALV6_POOL_RANGE_END
 - INITIAL_IRONICBRIDGE_IP -> INITIAL_BM_PROVISIONER_BRIDGE_IP
 - baremetal_network_cidr_v4 -> external_network_cidr_v4
 - baremetal_network_cidr_v6 -> external_network_cidr_v6
 - virsh_dhcp_v4_start -> external_dhcp_v4_start
 - virsh_dhcp_v4_end -> external_dhcp_v4_end
 - virsh_dhcp_v6_start -> external_dhcp_v6_start
 - virsh_dhcp_v6_end -> external_dhcp_v6_end
 - PROVISIONING_POOL_RANGE_START -> IPAM_PROVISIONING_POOL_RANGE_START
 - PROVISIONING_POOL_RANGE_END -> IPAM_PROVISIONING_POOL_RANGE_END
 - MANAGE_BR_BRIDGE -> MANAGE_EX_BRIDGE
 - every variable that had BM_ in it's name has BARE_METAL_ instead now

### CODE REVIEW

**Comment by Rozzii** [comment]
/hold 
I am testing, it is not done yet.

**Comment by mboukhalfa** [comment]
testing the new trigger format
/test-centos-e2e-integration-main

File: 04_verify.sh [hunk 1]
```diff
@@ -22,40 +22,40 @@ fi
 check_bm_hosts() {
     local FAILS_CHECK="${FAILS}"
     local NAME ADDRESS USER PASSWORD MAC CRED_NAME CRED_SECRET \
-      BM_HOSTS BM_HOST BM_VMS BM_VMNAME BM_VM_IFACES
+      BARE_METAL_HOSTS BARE_METAL_HOST BARE_METAL_VMS BARE_METAL_VMNAME BARE_METAL_VM_IFACES
```
Comment (tuminoid) [inline 04_verify.sh:25]: Rather than making it an extra long line, just have two lines of `local ...`

File: 04_verify.sh [hunk 2]
```diff
@@ -22,40 +22,40 @@ fi
...
+    BARE_METAL_HOSTS="$(kubectl --kubeconfig "${KUBECONFIG}" get baremetalhosts\
       -n metal3 -o json)"
-    BM_VMS="$(sudo virsh list --all)"
-    BM_VMNAME="${NAME//-/_}"
+    BARE_METAL_VMS="$(sudo virsh list --all)"
+    BARE_METAL_VMNAME="${NAME//-/_}"
     # Verify BM host exists
     RESULT_STR="${NAME} Baremetalhost exist"
-    echo "$BM_HOSTS" | grep -w "${NAME}"  > /dev/null
+    echo "$BARE_METAL_HOSTS" | grep -w "${NAME}"  > /dev/null
```
Comment (tuminoid) [inline 04_verify.sh:37]: Curlys missing.

File: 04_verify.sh [hunk 3]
```diff
@@ -71,20 +71,20 @@ check_bm_hosts() {
...
+    echo "$BARE_METAL_VMS "| grep -w "${BARE_METAL_VMNAME}"  > /dev/null
     process_status $?
 
     #Verify the VMs interfaces
-    BM_VM_IFACES="$(sudo virsh domiflist "${BM_VMNAME}")"
+    BARE_METAL_VM_IFACES="$(sudo virsh domiflist "${BARE_METAL_VMNAME}")"
     for bridge in ${BRIDGES}; do
       RESULT_STR="${NAME} Baremetalhost VM interface ${bridge} exist"
-      echo "$BM_VM_IFACES" | grep -w "${bridge}"  > /dev/null
+      echo "$BARE_METAL_VM_IFACES" | grep -w "${bridge}"  > /dev/null
```
Comment (tuminoid) [inline 04_verify.sh:81]: Curlys missing.

File: host_cleanup.sh [hunk 4]
```diff
@@ -59,7 +59,7 @@ if [[ $OS == "centos" || $OS == "rhel" ]]; then
   fi
   # Baremetal net should have been cleaned already at this stage, but we double
   # check as leaving it around causes issues when the host is rebooted
-  if [ "$MANAGE_BR_BRIDGE" == "y" ]; then
+  if [ "$MANAGE_EXT_BRIDGE" == "y" ]; then
```
Comment (tuminoid) [inline host_cleanup.sh:62]: `==` -> `=`

File: lib/images.sh [hunk 5]
```diff
@@ -24,8 +24,8 @@ else
 fi
 
 # Image url and checksum
-export IMAGE_URL=http://$PROVISIONING_URL_HOST/images/${IMAGE_NAME}
-export IMAGE_CHECKSUM=http://$PROVISIONING_URL_HOST/images/${IMAGE_NAME}.md5sum
+export IMAGE_URL=http://$BARE_METAL_PROVISIONER_URL_HOST/images/${IMAGE_NAME}
```
Comment (tuminoid) [inline lib/images.sh:27]: Curlys missing, and quote the right hand side.

File: lib/images.sh [hunk 6]
```diff
@@ -35,5 +35,5 @@ IMAGE_BASE_NAME="${IMAGE_NAME%.*}"
 export IMAGE_RAW_NAME="${IMAGE_BASE_NAME}-raw.img"
 
 # variables used for template parametrization in CAPM3
-export IMAGE_RAW_URL=http://$PROVISIONING_URL_HOST/images/${IMAGE_RAW_NAME}
-export IMAGE_RAW_CHECKSUM=http://$PROVISIONING_URL_HOST/images/${IMAGE_RAW_NAME}.md5sum
+export IMAGE_RAW_URL=http://$BARE_METAL_PROVISIONER_URL_HOST/images/${IMAGE_RAW_NAME}
```
Comment (tuminoid) [inline lib/images.sh:38]: Same.

File: lib/network.sh [hunk 7]
```diff
@@ -30,21 +30,21 @@ function prefixlen() {
 }
 
 # Provisioning Interface
-export CLUSTER_PROVISIONING_INTERFACE=${CLUSTER_PROVISIONING_INTERFACE:-"ironicendpoint"}
+export BARE_METAL_PROVISIONER_INTERFACE=${BARE_METAL_PROVISIONER_INTERFACE:-"ironicendpoint"}
```
Comment (tuminoid) [inline lib/network.sh:33]: Fix quotes to go around the assignment instead inside the braces.

File: lib/network.sh [hunk 8]
```diff
@@ -30,21 +30,21 @@ function prefixlen() {
...
 # POD CIDR
 export POD_CIDR=${POD_CIDR:-"192.168.0.0/18"}
 
 # Enables single-stack IPv6
-PROVISIONING_IPV6=${PROVISIONING_IPV6:-false}
+BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY=${BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY:-false}
 IPV6_ADDR_PREFIX=${IPV6_ADDR_PREFIX:-"fd2e:6f44:5dd8:b856"}
 
-if [[ "${PROVISIONING_IPV6}" == "true" ]]; then
+if [[ "${BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY}" == "true" ]]; then
```
Comment (tuminoid) [inline lib/network.sh:42]: `==` -> `=`

File: lib/network.sh [hunk 9]
```diff
@@ -30,21 +30,21 @@ function prefixlen() {
...
   export BOOT_MODE=${BOOT_MODE:-UEFI}
-  export PROVISIONING_NETWORK=${PROVISIONING_NETWORK:-fd2e:6f44:5dd8:b856::/64}
+  export BARE_METAL_PROVISIONER_NETWORK=${BARE_METAL_PROVISIONER_NETWORK:-fd2e:6f44:5dd8:b856::/64}
```
Comment (tuminoid) [inline lib/network.sh:44]: Add quotes.

File: lib/network.sh [hunk 10]
```diff
@@ -30,21 +30,21 @@ function prefixlen() {
...
 else
   export BOOT_MODE=${BOOT_MODE:-legacy}
-  export PROVISIONING_NETWORK=${PROVISIONING_NETWORK:-172.22.0.0/24}
+  export BARE_METAL_PROVISIONER_NETWORK=${BARE_METAL_PROVISIONER_NETWORK:-172.22.0.0/24}
```
Comment (tuminoid) [inline lib/network.sh:47]: Same.

File: lib/network.sh [hunk 11]
```diff
@@ -59,34 +59,36 @@ elif [[ "${BOOT_MODE}" == "UEFISecureBoot" ]]; then
 fi
 
 # shellcheck disable=SC2155
-prefixlen PROVISIONING_CIDR "$PROVISIONING_NETWORK"
-export PROVISIONING_CIDR
-export PROVISIONING_NETMASK=${PROVISIONING_NETMASK:-$(python -c "import ipaddress; print(ipaddress.ip_network(u\"$PROVISIONING_NETWORK\").netmask)")}
+prefixlen BARE_METAL_PROVISIONER_CIDR "$BARE_METAL_PROVISIONER_NETWORK"
+export BARE_METAL_PROVISIONER_CIDR
+export BARE_METAL_PROVISIONER_NETMASK=${BARE_METAL_PROVISIONER_NETMASK:-$(python -c "import ipaddress; print(ipaddress.ip_network(u\"$BARE_METAL_PROVISIONER_NETWORK\").netmask)")}
```
Comment (tuminoid) [inline lib/network.sh:64]: This has so many ways to fail and slip through, can we unpack it into an `if` statement please.

File: lib/network.sh [hunk 12]
```diff
@@ -59,34 +59,36 @@ elif [[ "${BOOT_MODE}" == "UEFISecureBoot" ]]; then
...
 
-network_address PROVISIONING_IP "$PROVISIONING_NETWORK" 1
-network_address CLUSTER_PROVISIONING_IP "$PROVISIONING_NETWORK" 2
+network_address BARE_METAL_PROVISIONER_IP "$BARE_METAL_PROVISIONER_NETWORK" 1
```
Comment (tuminoid) [inline lib/network.sh:66]: Entire lib is missing curlys, so I'm not going to complain about those.

File: lib/network.sh [hunk 13]
```diff
@@ -59,34 +59,36 @@ elif [[ "${BOOT_MODE}" == "UEFISecureBoot" ]]; then
...
-export PROVISIONING_IP
-export CLUSTER_PROVISIONING_IP
+export BARE_METAL_PROVISIONER_IP
+export CLUSTER_BARE_METAL_PROVISIONER_IP
 
 # shellcheck disable=SC2153
-if [[ "$PROVISIONING_IP" == *":"* ]]; then
-  export PROVISIONING_URL_HOST="[$PROVISIONING_IP]"
-  export CLUSTER_URL_HOST="[$CLUSTER_PROVISIONING_IP]"
+if [[ "$BARE_METAL_PROVISIONER_IP" == *":"* ]]; then
```
Comment (tuminoid) [inline lib/network.sh:73]: `==` -> `=`

File: lib/network.sh [hunk 14]
```diff
@@ -189,16 +191,16 @@ else
   export REGISTRY=${REGISTRY:-"[${EXTERNAL_SUBNET_V6_HOST}]:${REGISTRY_PORT}"}
 fi
 
-network_address INITIAL_IRONICBRIDGE_IP "$PROVISIONING_NETWORK" 9
+network_address INITIAL_BARE_METAL_PROVISIONER_BRIDGE_IP "$BARE_METAL_PROVISIONER_NETWORK" 9
 
-export DEPLOY_KERNEL_URL=${DEPLOY_KERNEL_URL:-"http://${CLUSTER_URL_HOST}:${HTTP_PORT}/images/ironic-python-agent.kernel"}
-export DEPLOY_RAMDISK_URL=${DEPLOY_RAMDISK_URL:-"http://${CLUSTER_URL_HOST}:${HTTP_PORT}/images/ironic-python-agent.initramfs"}
+export DEPLOY_KERNEL_URL=${DEPLOY_KERNEL_URL:-"http://${CLUSTER_BARE_METAL_PROVISIONER_HOST}:${HTTP_PORT}/images/ironic-python-agent.kernel"}
```
Comment (tuminoid) [inline lib/network.sh:196]: Move quotes around the whole assignment. Same for all lines below.

File: tests/roles/run_tests/templates/clusterctl-vars.yaml [hunk 15]
```diff
@@ -15,13 +15,13 @@ CLUSTER_APIENDPOINT_HOST: {{ CLUSTER_APIENDPOINT_HOST }}
...
-PROVISIONING_POOL_RANGE_START: {{ PROVISIONING_POOL_RANGE_START }}
-PROVISIONING_POOL_RANGE_END: {{ PROVISIONING_POOL_RANGE_END }}
-PROVISIONING_CIDR: {{ PROVISIONING_CIDR }}
-BAREMETALV4_POOL_RANGE_START: {{ BAREMETALV4_POOL_RANGE_START }}
-BAREMETALV4_POOL_RANGE_END: {{ BAREMETALV4_POOL_RANGE_END }}
-BAREMETALV6_POOL_RANGE_START: {{ BAREMETALV6_POOL_RANGE_START }}
-BAREMETALV6_POOL_RANGE_END: {{ BAREMETALV6_POOL_RANGE_END }}
+PROVISIONING_POOL_RANGE_START: {{ IPAM_PROVISIONING_POOL_RANGE_START }}
+PROVISIONING_POOL_RANGE_END: {{ IPAM_PROVISIONING_POOL_RANGE_END }}
+BARE_METAL_PROVISIONER_CIDR: {{ BARE_METAL_PROVISIONER_CIDR }}
```
Comment (tuminoid) [inline tests/roles/run_tests/templates/clusterctl-vars.yaml:20]: Is the `BARE_METAL` vs `BAREMETAL` inconsistency something we want to have?

File: ubuntu_bridge_network_configuration.sh [hunk 16]
```diff
@@ -20,10 +20,10 @@ if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
      # sudo ifconfig provisioning 172.22.0.1 netmask 255.255.255.0 up
      # Use ip command. ifconfig commands are deprecated now.
      sudo ip link set provisioning up
-     if [[ "${PROVISIONING_IPV6}" == "true" ]]; then
-        sudo ip -6 addr add "$PROVISIONING_IP"/"$PROVISIONING_CIDR" dev ironicendpoint
+     if [[ "${BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY}" == "true" ]]; then
```
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:23]: `==` -> `=`

File: ubuntu_bridge_network_configuration.sh [hunk 17]
```diff
@@ -20,10 +20,10 @@ if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
...
+        sudo ip -6 addr add "$BARE_METAL_PROVISIONER_IP"/"$BARE_METAL_PROVISIONER_CIDR" dev ironicendpoint
```
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:24]: Curlys.

File: ubuntu_bridge_network_configuration.sh [hunk 18]
```diff
@@ -20,10 +20,10 @@ if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
...
       else
-        sudo ip addr add dev ironicendpoint "$PROVISIONING_IP"/"$PROVISIONING_CIDR"
+        sudo ip addr add dev ironicendpoint "$BARE_METAL_PROVISIONER_IP"/"$BARE_METAL_PROVISIONER_CIDR"
```
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:26]: Curlys.

File: ubuntu_bridge_network_configuration.sh [hunk 19]
```diff
@@ -36,18 +36,18 @@ if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
  fi
 
  if [ "$MANAGE_INT_BRIDGE" == "y" ]; then
```
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:38]: `==` -> `=`

File: ubuntu_bridge_network_configuration.sh [hunk 20]
```diff
@@ -36,18 +36,18 @@ if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
...
-     # Create the baremetal bridge
-     if ! [[  $(ip a show baremetal) ]]; then
-       sudo brctl addbr baremetal
-       # sudo ifconfig baremetal 192.168.111.1 netmask 255.255.255.0 up
+     # Create the external bridge
+     if ! [[  $(ip a show external) ]]; then
```
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:40]: What is this supposed to be doing? It produces output but has no test.

Probably going for `if ! ip a show external &>/dev/null; then`  as a way to check if `external` exists?

File: ubuntu_bridge_network_configuration.sh [hunk 21]
```diff
@@ -58,9 +58,9 @@ if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
  fi
 
  # restart the libvirt network so it applies an ip to the bridge
- if [ "$MANAGE_BR_BRIDGE" == "y" ] ; then
-     sudo virsh net-destroy baremetal
-     sudo virsh net-start baremetal
+ if [ "$MANAGE_EXT_BRIDGE" == "y" ] ; then
```
Comment (tuminoid) [inline ubuntu_bridge_network_configuration.sh:61]: `==` -> `=`

**Review by tuminoid — COMMENTED** [review]
Some nits.

File: lib/images.sh (hunk 6 above)
Comment (Rozzii) [reply lib/images.sh:38]: done

File: 02_configure_host.sh [hunk 22]
```diff
@@ -60,7 +60,7 @@ else
       # dnsmasq being run, we don't want that as we have our own dnsmasq, so set
       # the IP address here
       if [ ! -e /etc/NetworkManager/system-connections/provisioning.nmconnection ] ; then
-        if [[ "${PROVISIONING_IPV6}" == "true" ]]; then
+        if [[ "${BARE_METAL_PROVISIONER_SUBNET_IPV6_ONLY}" == "true" ]]; then
```
[the remark this answers is not in the document]
Comment (Rozzii) [reply 02_configure_host.sh:63]: done

File: 04_verify.sh (hunk 1 above)
Comment (Rozzii) [reply 04_verify.sh:25]: done

File: 04_verify.sh (hunk 2 above)
Comment (Rozzii) [reply 04_verify.sh:37]: done

File: 04_verify.sh [hunk 23]
```diff
@@ -71,20 +71,20 @@ check_bm_hosts() {
 
     # Verify the VM was created
     RESULT_STR="${NAME} Baremetalhost VM exist"
-    echo "$BM_VMS "| grep -w "${BM_VMNAME}"  > /dev/null
+    echo "$BARE_METAL_VMS "| grep -w "${BARE_METAL_VMNAME}"  > /dev/null
```
[the remark this answers is not in the document]
Comment (Rozzii) [reply 04_verify.sh:74]: done

File: 04_verify.sh (hunk 3 above)
Comment (Rozzii) [reply 04_verify.sh:81]: done

File: host_cleanup.sh (hunk 4 above)
Comment (Rozzii) [reply host_cleanup.sh:62]: done

File: lib/images.sh (hunk 5 above)
Comment (Rozzii) [reply lib/images.sh:27]: done

File: lib/network.sh (hunk 7 above)
Comment (Rozzii) [reply lib/network.sh:33]: done

File: lib/network.sh (hunk 8 above)
Comment (Rozzii) [reply lib/network.sh:42]: done

File: lib/network.sh (hunk 10 above)
Comment (Rozzii) [reply lib/network.sh:47]: done

File: lib/network.sh (hunk 13 above)
Comment (Rozzii) [reply lib/network.sh:73]: done

File: lib/network.sh (hunk 14 above)
Comment (Rozzii) [reply lib/network.sh:196]: done

File: ubuntu_bridge_network_configuration.sh (hunk 17 above)
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:24]: done

File: ubuntu_bridge_network_configuration.sh (hunk 18 above)
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:26]: done

File: ubuntu_bridge_network_configuration.sh (hunk 19 above)
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:38]: done

File: ubuntu_bridge_network_configuration.sh (hunk 21 above)
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:61]: done

File: ubuntu_bridge_network_configuration.sh (hunk 16 above)
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh]: done

File: lib/network.sh (hunk 9 above)
Comment (Rozzii) [reply lib/network.sh]: done

File: tests/roles/run_tests/templates/clusterctl-vars.yaml (hunk 15 above)
Comment (Rozzii) [reply tests/roles/run_tests/templates/clusterctl-vars.yaml:20]: done, i have changed every BAREMETAL to BARE_METAL.

File: ubuntu_bridge_network_configuration.sh (hunk 20 above)
Comment (Rozzii) [reply ubuntu_bridge_network_configuration.sh:40]: yeah I think the original purpose of this condition was to check whether the external interface is present or not.

**Review by tuminoid — COMMENTED** [review]
Just more nits. I think these need cleanup in separate patch as all convention is completely random, and there are so many bad practices visible in these snippets. So feel free to fix/resolve any you feel like. No actual problems spotted with renaming.

File: lib/network.sh
[the remark this answers is not in the document]
Comment (Rozzii) [reply lib/network.sh:1]: I will fix curling and = signs in a separate refactoring PR I think most of the files in dev env have these incorrectly so I would prefer pushing a dedicated syntax fixing PR

[the remark this answers is not in the document]
Comment (tuminoid) [reply lib/network.sh:1]: That works for me. Thanks @Rozzii

**Comment by Rozzii** [comment]
Followup PR will implement shell syntax improvements mentioned by comments here.

**Review by lentzi90 — CHANGES_REQUESTED** [review]
Overall good improvements! It is now clearer what these variables do (but I must confess I still don't understand all of them :sweat_smile: ).

I'm a bit unsure if it is necessary to have `BARE_METAL` in all of them? For example, are there other provisioners than bare metal provisioners? Or could we just say `PROVISIONER` then instead of `BARE_METAL_PROVISIONER`? Just a thought :slightly_smiling_face: 

Upgrade vars should not be added back (see below)!

File: tests/feature_tests/upgrade/upgrade_vars.sh
[the remark this answers is not in the document]
Comment (Rozzii) [reply tests/feature_tests/upgrade/upgrade_vars.sh:1]: Thanks fort detecting this

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by tuminoid — DISMISSED** [review]
As agreed, my nits are addressed in separate PRs. There are some other pending changes so just signalling my approval for now.

File: vm-setup/roles/firewall/tasks/iptables.yaml [hunk 24]
```diff
@@ -110,13 +110,13 @@
         state: "{{ firewall_rule_state }}"
       loop: "{{ provisioning_host_ports }}"
 
-    - name: "iptables: Allow access to provisioning network from kind"
+    - name: "iptables: Allow access to bare metal provisioner network from kind"
```
Comment (mboukhalfa) [inline vm-setup/roles/firewall/tasks/iptables.yaml:113]: We are still keeping provisioning network name ?

File: vm-setup/library/generate_macs.py [hunk 25]
```diff
@@ -24,19 +24,19 @@
...
-   - Generate a list of Ethernet MAC addresses suitable for baremetal testing.
+   - Generate a list of Ethernet MAC addresses suitable for external testing.
 '''
 
 MAX_NUM_MACS = math.trunc(0xff / 2)
 
 
-def generate_baremetal_macs(nodes, networks):
-    """Generate an Ethernet MAC address suitable for baremetal testing."""
+def generate_external_macs(nodes, networks):
```
Comment (mboukhalfa) [inline vm-setup/library/generate_macs.py:33]: to me macs are not related to external network would prefer to name it something related to VM `generate_baremetal_vm_macs`

File: vm-setup/library/generate_macs.py [hunk 26]
```diff
@@ -24,19 +24,19 @@
...
+    """Generate an Ethernet MAC address suitable for external testing."""
     # NOTE(dprince): We generate our own bare metal MAC address's here
     # instead of relying on libvirt so that we can ensure the
     # locally administered bit is set low. (The libvirt default is
     # to set the 2nd MSB high.) This effectively allows our
-    # fake baremetal VMs to more accurately behave like real hardware
+    # fake external VMs to more accurately behave like real hardware
```
Comment (mboukhalfa) [inline vm-setup/library/generate_macs.py:39]: `external VMs` does not really make sense

File: vm-setup/library/generate_macs.py [hunk 27]
```diff
@@ -81,7 +81,7 @@ def main():
             networks=dict(required=True, type='list')
         )
     )
-    result = generate_baremetal_macs(module.params["nodes"],
+    result = generate_external_macs(module.params["nodes"],
```
Comment (mboukhalfa) [inline vm-setup/library/generate_macs.py:84]: check function definition comment

**Review by mboukhalfa — COMMENTED** [review]
few notes

File: vm-setup/roles/firewall/tasks/iptables.yaml (hunk 24 above)
Comment (Rozzii) [reply vm-setup/roles/firewall/tasks/iptables.yaml:113]: Yes, eventually there will be Provisioner and  Provisioning networks.  But in this specifc case of "{{ provisioning_host_ports }}" I have to check I might have missed this.

File: vm-setup/library/generate_macs.py (hunk 26 above)
Comment (Rozzii) [reply vm-setup/library/generate_macs.py]: done

File: vm-setup/library/generate_macs.py (hunk 27 above)
Comment (Rozzii) [reply vm-setup/library/generate_macs.py]: done

File: vm-setup/roles/firewall/tasks/iptables.yaml (hunk 24 above)
Comment (Rozzii) [reply vm-setup/roles/firewall/tasks/iptables.yaml:113]: I have changed "provisioning_host_ports" to "vm_host_ports"

**Review by tuminoid — APPROVED** [review]
/lgtm

**Comment by Rozzii** [comment]
Had to rebase looks like for some reason it was unable to get merged.
/test-centos-integration-release-main
/test-centos-integration-release-1-2
/test-ubuntu-e2e-integration-main
/test-centos-integration-release-1-3

**Pull request merged**